code/datums/components/holderloving.dm

```diff
@@ -0,0 +1,48 @@
+/// Prevents items from leaving their storage or whoever is holding their storage (does not need to have a storage component)
+/datum/component/holderloving
+	can_transfer = TRUE
+	/// Item that is kept track of
+	var/obj/item/holder
+	/// If we delete our parent when the holder gets deleted
+	var/del_parent_with_holder = FALSE
+
+/datum/component/holderloving/Initialize(_holder, _del_parent_with_holder)
```

Comment:
Make these `holder` and `del_parent_with_holder` instead, and do `src.holder = holder` instead of `holder = _holder`.